graylog2-web-interface/src/views/logic/SearchResult.js

```diff
@@ -1,18 +1,43 @@
-import _ from 'lodash';
+import { mapValues, get, compact } from 'lodash';
 import QueryResult from './QueryResult';
 import SearchError from './SearchError';
 
 class SearchResult {
   constructor(result) {
     this.result = result;
 
-    this.results = _.mapValues(result.results, queryResult => new QueryResult(queryResult));
-    this.errors = _.get(result, 'errors', []).map(error => new SearchError(error));
+    this.results = mapValues(result.results, queryResult => new QueryResult(queryResult));
+    this.errors = get(result, 'errors', []).map(error => new SearchError(error));
   }
 
   forId(queryId) {
     return this.results[queryId];
   }
+
+  updateSearchTypes(searchTypeResults) {
+    const updatedResult = new SearchResult(this.result).result;
+    searchTypeResults.forEach((searchTypeResult) => {
+      const searchQuery = this._getQueryBySearchTypeId(searchTypeResult.id);
+      updatedResult.results[searchQuery.query.id].search_types[searchTypeResult.id] = searchTypeResult;
+    });
+    return new SearchResult(updatedResult);
+  }
+
+  getSearchTypesFromResponse(searchTypeIds) {
+    const searchTypes = searchTypeIds.map((searchTypeId) => {
+      const relatedQuery = this._getQueryBySearchTypeId(searchTypeId);
+      return SearchResult._getSearchTypeFromQuery(relatedQuery, searchTypeId);
+    });
+    return compact(searchTypes);
```

Comment:
When there is an error due to the elastic search result window limit, the array of search types, created in line 27 will contain an `undefined`, because the api response does not contain the search type anymore, but the id is still defined in the `searchTypeIds` props. Right now we are just removing the undefined entires, to prevent any further errors.

In them message list we are showing an error message, for this case. My question is how to deal with this case properly. Is it okay to "ignore" the error here, because we are handling this case in the MessageList?

Reply:
Create here a function `filterFailedSearchTypes` where you do the `compat` then it is clear what you do. The place seems right to me.